src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,48 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int INITIAL_SIZE_CAPACITY = 10;
+    private static final int ADDING_SIZE = 5;
+    private T[] array;
+    private int size;
 
+    public MateStack() {
+        this.array = (T[]) new Object[INITIAL_SIZE_CAPACITY];
+        size = 0;
+    }
+
+    public void push(T value) {
+        if (size == array.length) {
+            array = grow(array);
+        }
+        array[size] = value;
+        size++;
     }
 
     public T peek() {
-        return null;
+        if (size == 0) {
+            throw new EmptyStackException();
+        }
+        return array[size - 1];
     }
 
     public T pop() {
-        return null;
+        T value = peek();
+        array[size - 1] = null;
+        size--;
+        return value;
     }
 
     public int size() {
-        return 0;
+        return size;
+    }
+
+    private T[] grow(T[] arrayValues) {
```

Comment:
```suggestion
    private void grow(T[] arrayValues) {
```
is an array changed in the whole application when it is modified inside some method?

Reply:
I think if I change the state of the array in method, then the original array will be modified as well.